app/models/transformation_mapping.rb

```diff
@@ -6,4 +6,72 @@ class TransformationMapping < ApplicationRecord
   def destination(source)
     transformation_mapping_items.find_by(:source => source).try(:destination)
   end
+
+  # vm_list: collection of hashes, each descriping a VM.
+  def validate_vms(vm_list = nil)
+    vm_list.present? ? identify_vms(vm_list) : select_vms
+  end
+
+  private
+
+  def select_vms
+    # TBD
+  end
+
+  def identify_vms(vm_list)
+    valid_list = []
+    invalid_list = []
+    conflict_list = []
+
+    vm_list.each do |row|
+      vm_name = row['Name']
+
+      query = Vm.where(:name => vm_name)
+      query = query.where(:uid_ems => row['UID']) if row['UID'].present?
+      query = query.where(:host => {:name => row['Host']}) if row['Host'].present?
+      query = query.where(:ext_management_system => {:name => row['Provider']}) if row['Provider'].present?
+
+      vms = query.to_a
+      invalid_list << describe_non_vm(vm_name) if vms.empty?
+
+      if vms.size == 1
+        reason = validate_vm(vms.first)
+        (reason.present? ? invalid_list : valid_list) << describe_vm(vms.first, reason)
+      else
+        vms.each { |vm| conflict_list << describe_vm(vm, N_("Duplicate")) }
+      end
+    end
+
+    {
+      "valid_vms"    => valid_list,
+      "invalid_vms"  => invalid_list,
+      "conflict_vms" => conflict_list
+    }
+  end
+
+  def describe_non_vm(vm_name)
+    {
+      "Name"   => vm_name,
+      "Reason" => N_("VM does not exist")
+    }
+  end
+
+  def describe_vm(vm, reason)
+    # formate to be finalized
+    {
+      "Name"           => vm.name,
+      "Cluster"        => vm.ems_cluster.name,
+      "Path"           => "#{vm.ext_management_system.name}/#{vm.parent_blue_folder_path(:exclude_non_display_folders => true)}",
+      "Allocated Size" => vm.allocated_disk_storage,
+      "ID"             => vm.id,
+      "Reason"         => reason
+    }
```

Comment:
My point was lost here in that these two methods are *all* presentation logic and belong outside of this model in something that deals with "csv" data.  I still don't agree that CSV is needed at all as that is a terrible transfer medium, particular at the REST API level, where an API user expects JSON.